benches/src/utils.rs

```diff
@@ -32,9 +32,104 @@ impl ShallowTempDir {
     }
 }
 
+const DB_CLEAN_UP_ENV_VAR: &str = "DB_CLEAN_UP";
+
 impl Drop for ShallowTempDir {
     fn drop(&mut self) {
-        // Ignore errors
-        let _ = std::fs::remove_dir_all(&self.path);
+        let default_db_clean_up = true;
+        // Check if DB_CLEAN_UP is set and correctly parsed to a boolean, defaulting to true
+        let should_clean_up = env::var(DB_CLEAN_UP_ENV_VAR)
+            .map_or(default_db_clean_up, |v| {
+                v.parse::<bool>().unwrap_or(default_db_clean_up)
+            });
+
+        if should_clean_up {
+            // Ignore errors when removing the directory
+            let _ = std::fs::remove_dir_all(&self.path);
+        }
+    }
+}
+
+#[cfg(test)]
+#[allow(non_snake_case)]
+mod tests {
+    use super::*;
+
+    fn shallow_temp_dir__drops_if_env_var_is_set() {
+        // given
+        env::set_var(DB_CLEAN_UP_ENV_VAR, "true");
+        let path;
+        {
+            let dir = ShallowTempDir::new();
+            path = dir.path().clone();
+            std::fs::create_dir_all(&path).expect("Failed to create temp directory");
```

Comment:
While unrelated to this PR, I would expect the `std::fs::create_dir_all` call to happen within `ShallowTempDir::new()`. Is there any scenario where we would want to create a `ShallowTempDir` without actually creating the directory?

Reply:
its just used in rocksdb, which automatically creates the required directories inside a directory ~ ShallowTempDir is really just a wrapper for a `PathBuf` which deletes anything at that path

Reply:
Ah alright, then it makes sense not to explicitly create the directory in the `new` function.Thanks for the context!